tools/validate-deployment.sh

```diff
@@ -0,0 +1,129 @@
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] Getting team_env directory failed" >> $ERROR_FILE
+    fi
+    # Getting stderr and stdout from custom script extension.
+    ssh -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress "bash -s" << EOF
+        sudo cp /var/lib/waagent/custom-script/download/0/stderr .;
+        sudo chown azureuser:azureuser ./stderr;
+EOF
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress:./stderr ./$teamAAD/
+    ssh -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress "bash -s" << EOF
+        sudo cp /var/lib/waagent/custom-script/download/0/stdout .;
+        sudo chown azureuser:azureuser ./stdout;
+EOF
+    errorflag=true
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress:./stdout ./$teamAAD/
+    # Getting deployment logs 
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE azureuser@$ipaddress:/home/azureuser/openhack-devops-proctor/provision-team/teamdeploy.out ./$teamAAD/
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] Getting teamdeploy.out file failed for subscription $subid, portal username $portalUserName, AAD $teamAAD, VM IP is $ipaddress" >> $ERROR_FILE
+        errorflag=false
+    fi
+
+    grep -x "poi   \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API poi has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE
+        errorflag=false 
+    fi
+
+    grep -x "user  \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API user has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE
+        errorflag=false 
+    fi
+
+    grep -x "trips \[X\]" ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API trips has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE
+        errorflag=false
+    fi
+
+    grep -x "user-java \[X\]" ./$teamAAD/teamdeploy.out
+     if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of API user-java has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE
+        errorflag=false 
+    fi
+    
+    grep -x '############ END OF TEAM PROVISION ############' ./$teamAAD/teamdeploy.out
+    if [ $? -ne 0 ]; then
+        echo "[ERROR] - Deployment of $teamAAD has failed in subscription $subid, portal username $portalUserName, AAD $teamAAD - VM IP is $ipaddress" >> $ERROR_FILE 
+        errorflag=false  
+    fi
+fi
+        
+# Packaging the results
+if [[ $ZIP_FILES ]]; then
+    zip -r teamfiles.zip OTA*
+    echo "Data from the teams deployment are in teamfiles.zip"
+fi
+
+if [ "$errorflag"=false ]; then 
```

Comment:
Are you sure this test is validation the value of the errorflag variable?

Reply:
@dcaro yes, i tested with sample bash script